tests/components/mqtt/test_climate.py

```diff
@@ -213,7 +213,7 @@ async def test_preset_modes_deprecation_guard(
     hass: HomeAssistant, caplog: pytest.LogCaptureFixture, parameter: str
 ) -> None:
     """Test the configuration for invalid legacy parameters."""
-    assert f"[{parameter}] is an invalid option for [mqtt]. Check: mqtt->mqtt->climate->0->{parameter}"
```

Comment:
Hello @emontnemery and @jbouwh, can you look at this test? It does nothing at all and I don't want to delete it, because I think it is supposed to test things. Thanks!

Reply:
It tests config in YAML for deprecated options. Probably this test is used for code coverage on some checks that are done.

Reply:
It seems to miss something to be a real test, because besides building the f-string nothing happens in there, it is not asserted against anything.

Reply:
It sets `hass_config`, a fixture that mocks configuration.yaml before `hass` is set up. The test asserts the correct working of the config validation.

Reply:
I have it now written so that it checks if the log contains the error messages.

Reply:
Why do we need to set up the entry if we only test the config validation?

Reply:
For me it looks like it was accidentally removed in #90105, as the other tests like `test_preset_none_in_preset_modes` have the `assert await mqtt_mock_entry()`.

https://github.com/home-assistant/core/pull/90105/files#diff-92f177e0bc75bf41e7cebaa920b6cf78e1ebd49c1d6101642cf7268b78316874L149-L157

Reply:
I'll have a closer look

Reply:
You are right. The assert does nothing. The whole test seems stale. I'll open a PR to remove it.

Reply:
https://github.com/home-assistant/core/pull/114443